packages/widget/src/stores/form/FormStore.tsx

```diff
@@ -10,44 +10,28 @@ import type { FormStoreStore } from './types.js';
 export const FormStoreProvider: React.FC<PropsWithChildren> = ({
   children,
 }) => {
-  const {
-    fromChain,
-    fromToken,
-    fromAmount,
-    toChain,
-    toToken,
-    toAddress,
-    hiddenUI,
-  } = useWidgetConfig();
+  const config = useWidgetConfig();
   const storeRef = useRef<FormStoreStore>();
 
-  const hiddenToAddress = hiddenUI?.includes(HiddenUI.ToAddress);
+  const hiddenToAddress = config.hiddenUI?.includes(HiddenUI.ToAddress);
 
   const defaultValues = useMemo(
     () => ({
       ...formDefaultValues,
-      fromChain,
-      fromToken,
-      toChain,
-      toToken,
+      fromChain: config.fromChain,
+      fromToken: config.fromToken,
+      toChain: config.toChain,
+      toToken: config.toToken,
       fromAmount:
-        (typeof fromAmount === 'number'
-          ? fromAmount?.toPrecision()
-          : fromAmount) || formDefaultValues.fromAmount,
+        (typeof config.fromAmount === 'number'
+          ? config.fromAmount?.toPrecision()
+          : config.fromAmount) || formDefaultValues.fromAmount,
       // Prevent setting address when the field is hidden
       toAddress: hiddenToAddress
         ? formDefaultValues.toAddress
-        : toAddress?.address || formDefaultValues.toAddress,
+        : config.toAddress?.address || formDefaultValues.toAddress,
     }),
-    [
-      fromAmount,
-      fromChain,
-      fromToken,
-      hiddenToAddress,
-      toAddress,
-      toChain,
-      toToken,
-    ],
+    [config, hiddenToAddress],
```

Comment:
There is a case that required this code to be changed. 

Before this change the issue could be recreated by doing the following, 
- In the playground if the user selected a from chain and token option from the design tools
- The user than changes the from chain and token in the widget itself to something else
- The user than tries to click the same from chain and token in the design tools as they did previously
- This last change would fail to update the from chain and token in the widget

I've change this code to look for general changes of the config object - so an immutable approach will allow those values to be updated. It might mean that this code is called more frequently though and we should discuss this approach.